tests/apidoc_login_test.go

```diff
@@ -85,7 +85,7 @@ func TestLogin(t *testing.T) {
 		t.Run("POST /login can log in as a user with just the local part of the id", func(t *testing.T) {
 			t.Parallel()
 
-			createDummyUser(t, unauthedClient, "local-login-user")
+			CreateDummyUser(t, unauthedClient, "local-login-user")
```

Comment:
We keep using this function, so let's move it to the main bit of Complement. Please move this to the `CSAPI` and `Deployment` structs. Something along the lines of:

```go
func (c *CSAPI) RegisterUser(t *testing.T, localpart, password string) (userID string, accessToken string) {
    // does the register request with the parameters given
    // returns the full user ID and access token, fails the test on network error
}
```
Which is then used in `Deployment`:
```go
func (d *Deployment) RegisterUser(t *testing.T, hsName, localpart, password string) *client.CSAPI {
    t.Helper()
    dep, ok := d.HS[hsName]
    if !ok {
       t.Fatalf("Deployment.Client - HS name '%s' not found", hsName)
       return nil
    }
    client := &client.CSAPI{
		BaseURL:          dep.BaseURL,
		Client:           client.NewLoggedClient(t, nil),
		SyncUntilTimeout: 5 * time.Second,
		Debug:            d.Deployer.debugLogging,
    }
    userID, accessToken := client.RegisterUser(t, localpart, password)

    // remember the token so subsequent calls to deployment.Client return the user
    dep.AccessTokens[userID] = accessToken

    client.UserID = userID
    client.AccessToken = accessToken
    return client
}
```
Which can then be used in tests:
```go
deployment := Deploy(t, b.BlueprintAlice)
charles := deployment.RegisterUser(t, "hs1", "charles", "mypassword")
```

Reply:
Hi @Kegsay , I did the changes and registered the method RegisterUser with deployment, however I am getting 400 whenever the client is making the POST request to the register endpoint. I have pushed the modified code, (do ignore the commented part, I will restore them later as soon as my util function gets fixed) Please do take a look at the helper functions registered!

Reply:
Done 😄 !